api/src/main/java/openconsensus/metrics/MetricOptions.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2019, OpenConsensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package openconsensus.metrics;
+
+import com.google.auto.value.AutoValue;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.concurrent.Immutable;
+import openconsensus.internal.Utils;
+import openconsensus.metrics.data.LabelKey;
+import openconsensus.metrics.data.LabelValue;
+
+/**
+ * Options for every metric added to the {@link MetricRegistry}.
+ *
+ * @since 0.1.0
+ */
+@Immutable
+@AutoValue
+public abstract class MetricOptions {
+
+  /**
+   * Returns the description of the Metric.
+   *
+   * <p>Default value is {@code ""}.
+   *
+   * @return the description of the Metric.
+   */
+  public abstract String getDescription();
+
+  /**
+   * Returns the unit of the Metric.
+   *
+   * <p>Default value is {@code "1"}.
+   *
+   * @return the unit of the Metric.
+   */
+  public abstract String getUnit();
+
+  /**
+   * Returns the list of label keys for the Metric.
+   *
+   * <p>Default value is {@link Collections#emptyList()}.
+   *
+   * @return the list of label keys for the Metric.
+   */
+  public abstract List<LabelKey> getLabelKeys();
+
+  /**
+   * Returns the map of constant labels (they will be added to all the TimeSeries) for the Metric.
+   *
+   * <p>Default value is {@link Collections#emptyMap()}.
+   *
+   * @return the map of constant labels for the Metric.
+   */
+  // TODO: add support for this and make it public.
```

Comment:
Is this todo copied or new?

Reply:
This is copied from https://github.com/census-instrumentation/opencensus-java/blob/a419d047b32a5c1dec661613a71f678544e0984d/api/src/main/java/io/opencensus/metrics/MetricOptions.java#L72.

/cc @bogdandrutu do you think we can expose this option and remove this todo?

Reply:
yes, please expose and make it public.